modules/security/pages/gsql-file-input-policy.adoc

```diff
@@ -0,0 +1,430 @@
+curl --user tigergraph:tigergraph  -d '
+[
+   {
+      "name":"load_person",
+      "dataSources":[
+         {
+            "filename":"f1",
+            "name":"k1",
+            "path":"",
+            "config":{
+               "topic":"kiwi",
+               "partition_list":[
+                  {
+                     "start_offset":-2,
+                     "partition":0
+                  }
+               ]
+            }
+         }
+      ],
+      "streaming":false
+   }
+]
+' -X POST "http://localhost:8123/gsql/loading-jobs?graph=test_graph&action=start"
+----
+
+==== CREATE LOADING JOB
+
+If a path is explicitly given (Ex. `sys.data_root`) when creating a loading job, users can check the path during creation of the loading job and block it immediately if not allowed.
+
+== User-Created Query
+
+=== Installed Mode
+
+==== selectVertex
+
+`selectVertex` eill read existing vertices from a local file directly.
+
+[NOTE]
+====
+Users should check filepath for the function.
+====
+
+[console,gsql]
+----
+CREATE QUERY selectVertexEx(STRING filename) FOR GRAPH socialNet {
+    S1 = {SelectVertex(filename, $"c1", $1, ",", true)};
+    S2 = {SelectVertex(filename, $0, person, ",", true)};
+    PRINT S1, S2; # Both sets of inputs product the same result
+}
+----
+
+==== LoadAccum
+
+`LoadAccum` is supported in a query to load data from local file into global accumulator.
+
+[console, gsql]
+----
+CREATE QUERY load_accum_ex (STRING filename) FOR GRAPH Social_Net {
+    TYPEDEF TUPLE<STRING aaa, VERTEX<Post> ddd> Your_Tuple;
+        MapAccum<VERTEX<Person>, MapAccum<INT, Your_Tuple>> @@test_map;
+        GroupByAccum<STRING a, STRING b, MapAccum<STRING, STRING> strList> @@test_group_by;
+
+        @@test_map = { LOADACCUM (filename, $0, $1, $2, $3, ",", false)};
+        @@test_group_by = { LOADACCUM ( filename, $1, $2, $3, $3, ",", true) };
+    PRINT @@test_map, @@test_group_by;
+}
+----
+
+== Path for Configurations
+We also allow `parse` and `read` configurations from local file system.
+These commands can be protected by file input policy as well, including:
+
+[console, gsql]
+----
+CREATE DATA_SOURCE KAFKA k1 = "/path/to/config"
+
+// from 3.9.0
+CREATE CONNECTOR FROM "/tmp/conn.cfg"
+CREATE DATA_SOURCE STREAM s1 = "/tmp/ds_config.json"
+----
+
+[NOTE]
+====
+If the object `DATA_SOURCE/CONNECTOR` is already created, and users can change the file input policy.
+Then the existing object won’t be affected because the config file is already read when creating the object.
+
+Also, `CREATE DATA_SOURCE` can only run with *local gsql client*, because the file is `read` from GSQL server.
+====
+
+== Put UDF file
+
+File input policy can also be applied to where UDF files are uploaded from.
+
+[console, gsql]
+----
+gsql '
+PUT ExprFunctions FROM "resources/gsql/common/ExprFunctions.hpp"
+'
+----
+
+[NOTE]
+====
+Similar to xref:_execute_gsql_file[], this restriction *does not* apply to remote GSQL client.
+It only apply to local GSQL client.
```

Comment:
Notice the `PUT` command can also use relative path (implicitly converted to absolute path with the GSQL client working directory)